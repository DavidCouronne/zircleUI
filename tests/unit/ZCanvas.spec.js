import { shallowMount, createLocalVue } from '@vue/test-utils'
import zcanvas from '@/components/z-canvas.vue'
import zircle from '@/index'
const localVue = createLocalVue()
localVue.use(zircle)
const wrapper = shallowMount(zcanvas, {
  localVue
})
describe('z-canvas.vue', () => {
  it('Renders class theme, color and AppMode', () => {
    expect(wrapper.find('#z-container').classes()).toContain('theme--dark')
    expect(wrapper.find('#z-container').classes()).toContain('color--blue')
    expect(wrapper.find('#z-container').classes()).toContain('full')
  })
  it('Has the expected html structure', () => {
    // Note: If props.imagesrc and or props.label are present, slots.imagesrc and slots.label wont be rendered
    expect(wrapper.element).toMatchSnapshot()
  })
})
