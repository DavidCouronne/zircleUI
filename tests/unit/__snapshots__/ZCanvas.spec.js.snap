// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-canvas.vue Has the expected html structure 1`] = `
<div
  class="theme--dark color--blue full"
  id="z-container"
>
  <div
    id="z-point"
  />
</div>
`;
